data.py

```diff
@@ -58,13 +59,153 @@ def jitter_pointcloud(pointcloud, sigma=0.01, clip=0.05):
     return pointcloud
 
 
+class Jitter:
+    def __init__(self, sigma=0.01, clip=0.05):
+        self._sigma = sigma
+        self._clip = clip
+
+    def __call__(self, pointcloud):
+        return jitter_pointcloud(pointcloud, self._sigma, self._clip)
+
+
+class RandomRemove:
+    """
+    Remove pointcloud randomly with an arbitrary probability.
+    """
+    def __init__(self, prob=0.2):
+        self._prob = prob
+
+    def __call__(self, pointcloud):
+        original_length = pointcloud.shape[0]
+        random_uniform = np.random.uniform(0, 1, pointcloud.shape[0])
+        masked_in = random_uniform > 0.2
+
+        masked_pointcloud = pointcloud[masked_in, :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveFourQuadrant:
+    def __call__(self, pointcloud):
+
+        original_length = pointcloud.shape[0]
+        along_which_axis = np.random.randint(3)
+
+        boolean_arr_masked_out = np.ones(original_length)
+        for i in range(3):
+            if i == along_which_axis:
+                continue
+            is_larger = np.random.randint(2)
+            boolean_arr_axis = pointcloud[:, i] > 0 if is_larger else pointcloud[:, i] < 0
+            boolean_arr_masked_out = np.logical_and(boolean_arr_axis, boolean_arr_masked_out)
+
+        masked_pointcloud = pointcloud[np.logical_not(boolean_arr_masked_out), :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveTwoQuadrant:
+    def __call__(self, pointcloud):
+        original_length = pointcloud.shape[0]
+        along_which_axis = np.random.randint(3)
+
+        is_larger = np.random.randint(2)
+        boolean_mask = pointcloud[:, along_which_axis] > 0 if is_larger else pointcloud[:, along_which_axis] < 0
+
+        masked_pointcloud = pointcloud[boolean_mask, :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveEightQuadrant:
```

Comment:
This is the class that removes an eighth of the pcds during training and validation


@siddhantwadhwa will be in charge of the experiment and validation